nilearn/input_data/confounds_parser.py

```diff
@@ -0,0 +1,305 @@
+from .confounds_utils import (
+    _sanitize_confounds,
+    _confounds_to_df,
+    _prepare_output,
+    MissingConfound,
+)
+from . import confounds_components as components
+
+
+# Global variables listing the admissible types of noise components
+all_confounds = [
+    "motion",
+    "high_pass",
+    "wm_csf",
+    "global",
+    "compcor",
+    "ica_aroma",
+    "scrub",
+    "non_steady_state",
+]
+
+# extra parameters needed for each noise component
+component_parameters = {
+    "motion": ["motion"],
+    "wm_csf": ["wm_csf"],
+    "global": ["global_signal"],
+    "compcor": ["meta_json", "compcor", "n_compcor"],
+    "ica_aroma": ["ica_aroma"],
+    "scrub": ["scrub", "fd_thresh", "std_dvars_thresh"],
+}
+
+
+def _sanitize_strategy(strategy):
+    """Define the supported denoising strategies."""
+    if not isinstance(strategy, list):
+        raise ValueError("strategy needs to be a list of strings")
+    for conf in strategy:
+        if conf not in all_confounds:
+            raise ValueError(f"{conf} is not a supported type of confounds.")
+    # add non steady state if not present
+    if "non_steady_state" not in strategy:
+        strategy.append("non_steady_state")
+    return strategy
+
+
+def _check_error(missing):
+    """Consolidate a single error message across multiple missing confounds."""
+    if missing["confounds"] or missing["keywords"]:
+        error_msg = (
+            "The following keywords or parameters are missing: "
+            + f" {missing['confounds']}"
+            + f" {missing['keywords']}"
+            + ". You may want to try a different denoising strategy."
+        )
+        raise ValueError(error_msg)
+
+
+def load_confounds(img_files,
+                   strategy=["motion", "high_pass", "wm_csf"],
+                   motion="full",
+                   scrub="full", fd_thresh=0.2, std_dvars_thresh=3,
+                   wm_csf="basic",
+                   global_signal="basic",
+                   compcor="anat_combined", n_compcor="auto",
+                   ica_aroma="full",
+                   demean=True):
+    """
+    Use confounds from fmriprep.
+
+    To enable easy confound variables loading from fMRIprep outputs, Confounds
+    provides a interface that groups subsets of confound variables into noise
+    components and their parameters. It is possible to fine-tune a subset of
+    noise components and their parameters through Confounds.
+
+    The implementation will only support fMRIPrep output from the 1.2.x series.
+    The `compcor` noise component requires 1.4.x series or above.
+
+    Parameters
+    ----------
+    img_files : path to processed image files, optionally as a list.
+        Processed nii.gz/dtseries.nii/func.gii file from fmriprep.
+        `nii.gz` or `dtseries.nii`: path to files, optionally as a list.
+        `func.gii`: list of a pair of paths to files, optionally as a list
+        of lists. The companion tsv will be automatically detected.
```

Comment:
Isn't there any way to document properly how the  tsv file **should** be organized, beyond "being the one output by fmriprep" ? If there is such a strong adherence to fmriprep that it becomes futile to describe the input file, then we should explictcly relate to the fmriprep dependency in the name of the function.

Reply:
This is a good point. We technically didn't write the test on a full fmriprep functional derivative output layout. As long as the image file and the associated confound tsv are in the same directory, things are fine. Definitely should add more details here.